### PR TITLE
Add functionality to filter instances through labels

### LINKED ISSUE #5710: in the service instance view, make it possible to click on the labels or pie chart to set the filter
image

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

on the video when clicking on "warning" and "info" labels it seems like nothing happens, or the filters are removed, it's due to the lack of the labels in the service that corresponds to given category.

closes #5710 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Code is clear and sufficiently documented
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

https://github.com/inmanta/web-console/assets/113331659/e93d3d35-26ff-490e-be1f-659f6563b086

### CODE REVIEW

**Comment by LukasStordeur** [comment]
As much as I like native JS approaches, it doesn't align much with how the app is working. Using React and co, I believe using the document to add eventListeners goes against the best practices. I get we use it when we don't have an option, or if we need to work around React, but this is not the case now. 

Why not use the context provider of the Service Inventory and sharing it with the chart? 

Also, I'm not a fan of having the "empty" states doing nothing.  I'd at least expect the chips to be added or have some sort of user feedback instead of a blank action.

A tooltip might be interesting to indicate that there's an action there.

**Comment by matborowczyk** [comment]
> As much as I like native JS approaches, it doesn't align much with how the app is working. Using React and co, I believe using the document to add eventListeners goes against the best practices. I get we use it when we don't have an option, or if we need to work around React, but this is not the case now.
> 
> Why not use the context provider of the Service Inventory and sharing it with the chart?
> 
> Also, I'm not a fan of having the "empty" states doing nothing. I'd at least expect the chips to be added or have some sort of user feedback instead of a blank action.
> 
> A tooltip might be interesting to indicate that there's an action there.

Right, I didn't think of the context, somehow I forgot about it totally. using event listeners was was simplest and "fastest" way to get the data from there to filters but I'll change that.

I am aware about empty states being an issue, as I forgot about context, the data that I had didn't allow me to know beforehand about whether the service have any filters that fit under this category.

I'll move those into some local context to provide better and two-way communication between those two places.

**Review by LukasStordeur — COMMENTED** [review]
I'm confused, why does warning and info have no filters? Some others do have filters when there are no matching instances. This behavior feels buggy to me. 

Perhaps this feature should have been discussed further before development

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx [hunk 1]
```diff
@@ -75,54 +94,72 @@ export const ServiceInventory: React.FunctionComponent<{
...
-                  instances={instances}
-                  serviceEntity={service}
-                  sort={sort}
-                  setSort={setSort}
-                />
-              ) : (
-                <EmptyView
-                  message={words("inventory.empty.message")(serviceName)}
-                  aria-label="ServiceInventory-Empty"
+    <LabelContext.Provider
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/ServiceInventory.tsx]: Good point, I merged them together

**Comment by matborowczyk** [comment]
https://github.com/inmanta/web-console/assets/113331659/25b4201a-5674-46eb-b685-7717f46fe481

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx [hunk 2]
```diff
@@ -43,6 +42,31 @@ const PreppedServiceInventory: React.FC<{ service: ServiceModel }> = ({
   />
 );
 
+interface ServiceInventoryContextProps {
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.tsx:45]: any objections to put the labels into an object? Feels like it make sense to group them to me.

Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/ServiceInventory.tsx:45]: grouped

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx [hunk 3]
```diff
@@ -41,7 +41,19 @@ function setup() {
...
+          <ServiceInventoryContext.Provider
+            value={{
+              labels: {
+                danger: [],
+                warning: [],
+                success: [],
+                info: [],
+                no_label: [],
+              },
+              onClick: jest.fn(),
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx:53]: is the onClick not part of the labels logic?

Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx:53]: It's filtering callback, so it uses labels but per se I would keep it separated, I can change the name though to be more descriptive

Comment (LukasStordeur) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx:53]: yes, because now it's not clear what the onClick does or with what it belongs. And for me it relates to the labels, it's when you click on a label/graph section.

Comment (LukasStordeur) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx:53]: if it makes more sense you can also rename the group to : `labelFiltering`.

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.tsx:1]: Can you also add new-lines between the different sections? Makes it a bit more readable and coherent :D

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx [hunk 4]
```diff
@@ -43,6 +42,35 @@ const PreppedServiceInventory: React.FC<{ service: ServiceModel }> = ({
...
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.tsx:45]: Props if these are local props.

File: src/Slices/ServiceInventory/UI/GetInstancesContext.tsx
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/GetInstancesContext.tsx:1]: Why did you remove it instead of adding the labels to it?  I don't mind the name change if we keep it consistent.

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx (hunk 3 above)
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.test.tsx]: changed

File: src/Slices/ServiceInventory/UI/GetInstancesContext.tsx
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/GetInstancesContext.tsx:1]: Actually I have no idea, I think it was faster to add refetch to the "ServiceInventoryContext" than other way around as I made them at first independent and separate. That's my guess after holiday :D

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx [hunk 5]
```diff
@@ -43,6 +42,35 @@ const PreppedServiceInventory: React.FC<{ service: ServiceModel }> = ({
...
+      warning: [],
+      success: [],
+      info: [],
+      no_label: [],
+    },
+    onClick: (_label) => {
+      // Default to no-op
+      return;
+    },
+    refetch: () => {
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.tsx]: why not like this, that's how it was in the initial Context
```
onClick: (_label) => null;
refetch: () => null;
```

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.tsx:1]: Still missing some comments in this file :)

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx (hunk 5 above)
Comment (LukasStordeur) [reply src/Slices/ServiceInventory/UI/ServiceInventory.tsx]: my bad, I'll place it on the right lines again, because it still applies ;)

File: src/Slices/ServiceInventory/UI/ServiceInventory.tsx [hunk 6]
```diff
@@ -43,6 +42,37 @@ const PreppedServiceInventory: React.FC<{ service: ServiceModel }> = ({
...
+}
+
+export const ServiceInventoryContext = createContext<Props>({
+  labelFiltering: {
+    danger: [],
+    warning: [],
+    success: [],
+    info: [],
+    no_label: [],
+    onClick: (_label) => {
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.tsx:64]: onClick: (_label) => null;
refetch: () => null;

**Review by LukasStordeur — APPROVED** [review]
